src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClones = [];
+
+  for (const action of actions) {
+    const currentState = {};
+
+    if (stateClones.length !== 0) {
+      Object.assign(currentState, stateClones[stateClones.length - 1]);
+    }
+
+    if (stateClones.length === 0) {
```

Comment:
The condition 'if (stateClones.length === 0)' is not necessary. You can use Object.assign() directly to copy the state. This will simplify the code.

Reply:
Not resolved from the previous review